todo app/src/store/todo.js

```diff
@@ -0,0 +1,34 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const todoSlice = createSlice({
+  name: "Todo",
+  initialState: [
+    {
+      id: "0",
+      task: "IAmHere",
+      done: false,
+    },
+  ],
+  reducers: {
+    add(state, action) {
+      const text = action.payload.trim();
+      if (text.length === 0) return;
+      const todo = { id: state.length, task: text, done: false };
+      state.push(todo);
+    },
+
+    delete(state, action) {
+      return state.filter((todo) => todo.id !== action.payload);
+    },
+
+    changeTaskStatus(state, action) {
+      const todoIndex = state.findIndex((todo) => todo.id === action.payload);
+      const todo = state[todoIndex];
```

Comment:
If you have the todo ID why not find the todo itself using `state.find()` it will save you one line but I guess that counts 😁

Reply:
The reason I found the index is because I wanted to mutate the object on that index otherwise I would not be able to mutate the todo I'm looking for.

Reply:
Fixed